patroni/config_generator.py

```diff
@@ -0,0 +1,289 @@
+"""patroni --generate-config machinery."""
+import os
+import socket
+import sys
+import yaml
+
+from typing import Any, Dict, Optional
+
+from .config import Config
+from .exceptions import PatroniException
+from .postgresql.config import ConfigHandler
+from .postgresql.misc import postgres_major_version_to_int
+from .utils import get_major_version, patch_config
+
+
+def get_ip() -> str:
+    """
+    Try to get an ip address of the hostname returned by :func:`gethostname`.
+
+    .. note::
+            Can also return local ip
```

Comment:
Maybe it's worth adding a note that this function can quit execution when facing `OSError` or `IndexError` exceptions.

Reply:
hm, maybe execution quitting actually should not happen here...